wait-for-statuses.py

```diff
@@ -0,0 +1,36 @@
+import urllib.request
```

Comment:
I'm not really sure about how this is to be used in the wild, but I would blindly argue against a) being hardly maintainable. Sometimes it's better not to fight against the environment, despite it looking poorly featured. Using `needs` in GitHub Actions workflows allows having nice graphs for understanding complex setups. See, for instance https://github.com/ghdl/ghdl/actions/runs/475367314 or https://github.com/verilator/verilator/actions/runs/476444622.

You can handle several dependencies into a single `needs` value by using matrices. So, I would try to look at "jobs" as stages/groups and expand the matrices dynamically if you want to avoid manually maintaining lists of tasks. By doing so, `wait-for-statuses.py` would only be required for checking the statuses of external services (say kokoro).

Reply:
Thank you for your insight. In Travis CI we used that "stages" approach, where jobs were grouped together. In GitHub Actions we wanted to have these precise dependency relations between jobs. I guess we'll stick with list of tasks for now.